onnxscript/type_annotation.py

```diff
@@ -146,3 +164,102 @@ def get_return_types(typeinfo: type | typing.Sequence[type]) -> typing.Sequence[
     if get_origin(typeinfo) is tuple:
         return get_args(typeinfo)
     return (typeinfo,)
+
+
+# A sorted list of all type strings used in an OpSchema
+ALL_TYPE_STRINGS = (
+    "tensor(bfloat16)",
+    "tensor(bool)",
+    "tensor(double)",
+    "tensor(float)",
+    "tensor(float16)",
+    "tensor(int16)",
+    "tensor(int32)",
+    "tensor(int64)",
+    "tensor(int8)",
+    "tensor(string)",
+    "tensor(uint16)",
+    "tensor(uint32)",
+    "tensor(uint64)",
+    "tensor(uint8)",
+)
+
+
+def pytype_to_input_strings(pytype: TypeAnnotationValue) -> list[str]:
+    """Returns a list of all supported input types in string representation for a given type annotation.
+
+    Args:
+        pytype: A type annotation.
+
+    Returns:
+        A list of all supported input types for the given type annotation.
+        Ensures that the list is sorted in the same order as ALL_TYPE_STRINGS.
+    """
+    if pytype is None:
+        return list(ALL_TYPE_STRINGS)
+    if pytype is type(None):
+        return list(ALL_TYPE_STRINGS)
+    if pytype is onnx_types.TensorType:
+        return list(ALL_TYPE_STRINGS)
+    if isinstance(pytype, type) and issubclass(pytype, onnx_types.TensorType):
+        return [pytype.to_string()]
+    if isinstance(pytype, onnx_types.TensorType):
+        return [pytype.to_string()]
+    if isinstance(pytype, TypeVar):
+        constraints = pytype.__constraints__
+        if constraints:
+            return pytype_to_input_strings(Union.__getitem__(constraints))
+        bound = pytype.__bound__
+        if bound is None:
+            return list(ALL_TYPE_STRINGS)
+        return pytype_to_input_strings(bound)
+    if typing.get_origin(pytype) is Union:
+        options = []
+        subtypes = typing.get_args(pytype)
+        # A None type in a Union is equivalent to an optional type
+        is_optional = any(subtype is type(None) for subtype in subtypes)
```

Comment:
We need some annotation to distinguish the two kinds of optionals ONNX supports.

Reply:
For now I am just mentally treating everything to be static optional. Any suggestions?